### PR TITLE
Currency as interface for FiatCurrency and Token(incl. NativeToken)

### PR DESCRIPTION
**Pull request by newtectonics**
Currency dto, can be used to input several types of currency into price.

### CODE REVIEW

**Comment by leafyoung** [comment]
Beside `0xeeee...` for ETH, where could BTC `0xbBbBBB...` be implemented?

**Comment by leafyoung** [comment]
> Beside `0xeeee...` for ETH, where could BTC `0xbBbBBB...` be implemented?

in Token

File: credmark/cmf/types/token.py [hunk 1]
```diff
@@ -23,8 +25,8 @@ def get_token_from_configuration(
     token_datas = [
         t for t in FUNGIBLE_TOKEN_DATA.get(chain_id, [])
```
Comment (iwyrkore) [inline credmark/cmf/types/token.py:26]: @leafyoung Shouldn't the FUNGIBLE_TOKEN_DATA be indexed into 2 dicts for looking up by symbol or address like we do for FIAT_CURRENCY_DATA_BY_ADDRESS? That would save us from looping through all the tokens.

Comment (iwyrkore) [reply credmark/cmf/types/token.py:26]: @leafyoung Do you want to do this? ^

Comment (leafyoung) [reply credmark/cmf/types/token.py:26]: yup. I will do it. running the models test now.

Comment (leafyoung) [reply credmark/cmf/types/token.py]: I made a commit before my flight across the pacific. I will be back online on Mon evening (CDT).

**Review by iwyrkore — APPROVED** [review]

**Pull request merged**